frontend/app/professor/test-simulations/page.tsx

```
@@ -1417,309 +1817,603 @@ ${availablePersonas.map(persona => `• @${persona.name.toLowerCase().replace(/\
+                          {(message.type === 'orchestrator' && (message as any).showLoadingBar) || (message as any).gradingInProgress || (message as any).sceneLoading ? (
+                              <>
+                                <div className="text-sm text-gray-600 mb-1">{(message as any).sceneLoading ? 'Loading next scene...' : (message as any).gradingInProgress ? 'Submitting for grading...' : 'Simulation is loading...'}</div>
+                                <div className="w-full mt-1 h-2 bg-gray-200 rounded-full overflow-hidden">
+                                  <div className="h-2 bg-blue-400 animate-pulse w-full transition-all duration-1000" style={{ width: '100%' }}></div>
+                                </div>
+                              </>
+                            ) : (
+                              message.text.split('\n').map((line, index) => {
+                                const boldFormatted = line.replace(/\*\*(.*?)\*\*/g, '<strong>$1</strong>')
+                                return (
+                                  <div key={index} dangerouslySetInnerHTML={{ __html: boldFormatted }} />
```

Biome (2.1.2)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)